server/src/main/java/io/unitycatalog/server/service/ModelService.java

```diff
@@ -3,102 +3,260 @@
     return HttpResponse.ofJson(createModelVersionResponse);
   }
 
   @Get("/{full_name}/versions")
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #registered_model, OWNER) ||
+          (#authorize(#principal, #registered_model, EXECUTE) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
   public HttpResponse listModelVersions(
-      @Param("full_name") String fullName,
+      @Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullName,
       @Param("max_results") Optional<Integer> maxResults,
       @Param("page_token") Optional<String> pageToken) {
     return HttpResponse.ofJson(MODEL_REPOSITORY.listModelVersions(fullName, maxResults, pageToken));
   }
 
   @Get("/{full_name}/versions/{version}")
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #registered_model, OWNER) ||
+          (#authorize(#principal, #registered_model, EXECUTE) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
   public HttpResponse getModelVersion(
-      @Param("full_name") String fullName, @Param("version") Long version) {
+      @Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullName, @Param("version") Long version) {
     assert fullName != null && version != null;
     ModelVersionInfo modelVersionInfo = MODEL_REPOSITORY.getModelVersion(fullName, version);
     return HttpResponse.ofJson(modelVersionInfo);
   }
 
   @Patch("/{full_name}/versions/{version}")
-  public HttpResponse updateModelVersion(UpdateModelVersion updateModelVersion) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #catalog, OWNER) ||
+          (#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse updateModelVersion(@Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullName,
+                                         UpdateModelVersion updateModelVersion) {
     assert updateModelVersion != null;
     ModelVersionInfo updateModelVersionResponse =
         MODEL_REPOSITORY.updateModelVersion(updateModelVersion);
     return HttpResponse.ofJson(updateModelVersionResponse);
   }
 
   @Delete("/{full_name}/versions/{version}")
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
```

Comment:
Catalog and schema owners should be able to delete a model version.

Reply:
Done.